agent/metrics/instancehealth/interface.go

```diff
@@ -0,0 +1,30 @@
+package instancehealth
+
+// InstanceHealth interface defines the behaviour of any counter
+// that uses genericCounter to collect the metrics used to determine
+// instance's health
+// As a starting point we will use this interface to collect Docker API
+// metrics. In future it can be extended to different counters.
+type InstanceHealth interface {
+
+	// This function increments the API call count
+	IncrementCallCount()
+
+	// This function returns the API call count
+	GetCallCount()
+
+	// This function increments the API's error count
+	IncrementErrorCount()
+
+	// This function returns the API error count
+	GetErrorCount()
+
+	// Records the error message
```

Comment:
in the comment add that it also increments error count

Reply:
Done.